test/CSharpIsNullAnalyzer.Tests/CSIsNull002Tests.cs

```diff
@@ -177,4 +177,66 @@ void Method(object o)
         await VerifyCS.VerifyCodeFixAsync(source, fixedSource1, CSIsNull002Fixer.IsObjectEquivalenceKey);
         await VerifyCS.VerifyCodeFixAsync(source, fixedSource2, CSIsNull002Fixer.IsNotNullEquivalenceKey);
     }
+
+    [Fact]
+    public async Task NotEqualsNullInExpressionTree_TargetTyped_ProducesNoDiagnostic()
+    {
+        string source = @"
+using System;
+using System.Linq.Expressions;
+
+class Test
+{
+    void Method()
+    {
+        Expression<Func<string, bool>> e = s => s [|!= null|];
+    }
+}";
+
+        string fixedSource = @"
+using System;
+using System.Linq.Expressions;
+
+class Test
+{
+    void Method()
+    {
+        Expression<Func<string, bool>> e = s => s is object;
+    }
+}";
+
+        await VerifyCS.VerifyCodeFixAsync(source, fixedSource, CSIsNull002Fixer.IsObjectEquivalenceKey);
+        // TODO: assert that a IsNotNullEquivalenceKey is NOT offered, as that would produce a compilation error.
```

Comment:
@sharwell How do I use your analyzer framework to assert that a particular code fix (by equivalence key) is _not_ offered?

Reply:
Set `CodeActionEquivalenceKey` to the item which should not be offered, and set `TestCode` and `FixedCode` to the same content. When `TestCode` and `FixedCode` are the same, the default number of expected iterations is 0, which means no matching code action is presented. Setting `CodeActionEquivalenceKey` filters the candidate actions:

https://github.com/dotnet/roslyn-sdk/blob/e5420e26d42234bb189aa43f093fc7d79e2a2bee/src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/CodeActionTest%601.cs#L138

Reply:
Thanks.